app/models/spree/order.rb

```diff
@@ -743,5 +745,22 @@ def set_payment_amount!
 
       pending_payments.first.update_attribute :amount, total
     end
+
+    def save_default_addresses
+      return unless save_bill_address || save_ship_address
+
+      if save_bill_address
+        customer.bill_address_id = bill_address_id
+        user&.bill_address_id = bill_address_id
+      end
+
+      if save_ship_address
+        customer.ship_address_id = ship_address_id
+        user&.ship_address_id = ship_address_id
+      end
+
+      customer.save
```

Comment:
Why do we save the customer? I don't see where we change it.

Reply:
We set `customer.bill_address_id` / `customer.ship_address_id` above.

Reply:
:facepalm: 

You see! The complexity was too much! :duck: 

Somehow I didn't see the `.` and though that you were just assigning a temporary variable.